src/thunderbots/software/backend_output/grsim/grsim_backend.cpp

```diff
@@ -58,12 +63,46 @@ void GrSimBackend::sendGrSimPacket(const grSim_Packet& packet)
         remote_endpoint, 0, err);
 }
 
+
 void GrSimBackend::sendPrimitives(
     const std::vector<std::unique_ptr<Primitive>>& primitives)
 {
+    std::vector<grSim_Packet> grsim_packets;
+
+    MotionController::Velocity robot_velocities;
+    grSim_Packet grsim_packet;
+
+
+    // get timestamp before running bang-bang controller for all robots
+    auto bangbang_timestamp = std::chrono::steady_clock::now();
+
+    std::chrono::duration<double> delta_time;
+
     // TODO: Implement this
-    // https://github.com/UBC-Thunderbots/Software/issues/21
-    grSim_Packet grsim_packet = createGrSimPacket(
-        0, YELLOW, Point(0.5, -0.1), Angle::ofRadians(-0.8), 4.0, true, false);
+    // https://github.com/UBC-Thunderbots/Software/issues/99
+    for (auto& prim : primitives)
+    {
+        MovePrimitive movePrim = dynamic_cast<MovePrimitive&>(*prim);
+
+        auto current_time =
+            std::chrono::steady_clock::now();  // get the current time right before
+                                               // running bang-bang to get a time-delta
+                                               // for acceleration
+
+        delta_time = current_time - bangbang_timestamp;
+
+        // Motion controller determines the speeds to send to the robots based on their
+        // state, maximum linear and angular accelerations, and the robots target
+        // destination location/orientation and speed
+        MotionController::bangBangVelocityController(
```

Comment:
The returned value is not being used, and `robot_velocities` is not being updated.

Reply:
Good catch. Now updates robot_velocities.